### PR TITLE
Allow NPC view data modifications

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #4289

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Updated script commands setunitdata and getunitdata to support the modification of NPC view data.
  * Converted mob_avail database to YAML.
Thanks to @4144, @exneval, and @Balferian!

### CODE REVIEW

**Comment by Balferian** [comment]
If npc have sprite = 0 without setunitdata UNPC_CLASS it looks like a novice.
Is this correct?
`prontera,155,175,4	script	NPCs	0,4,4,{ end; }`
image

**Comment by aleos89** [comment]
Novice is ID 0, so that sounds right. UNPC_CLASS doesn't just mean _job class_, it refers to the general display of the NPC.

**Comment by aleos89** [comment]
A client crash or map crash?

**Comment by Balferian** [comment]
image
I think he say about novice sprite.

**Comment by attackjom** [comment]
for me these novice are headless and pop-up client crash 

so , maybe it a client-side problem

**Comment by Balferian** [comment]
Need to change function in **map/npc.cpp**:
```
struct npc_data *npc_create_npc(int m, int x, int y)
{
	struct npc_data *nd = NULL;

	CREATE(nd, struct npc_data, 1);
	nd->bl.id = npc_get_new_npc_id();
	nd->bl.prev = nd->bl.next = NULL;
	nd->bl.m = m;
	nd->bl.x = x;
	nd->bl.y = y;
	nd->area_size = AREA_SIZE + 1;
	nd->sc_display = NULL;
	nd->sc_display_count = 0;
	nd->progressbar.timeout = 0;
	nd->vd = npc_viewdb[0];

	return nd;
}
```

**Comment by Balferian** [comment]
When use **setunitdata getnpcid(0),UNPC_BODY2,1;** map server crashed.

**Comment by aleos89** [comment]
Could you provide your crash dump? Thanks.

**Comment by Balferian** [comment]
Sample script:
prontera,158,172,4	script	NPCs	4_NFWISP,{
	setunitdata getnpcid(0),UNPC_CLASS,4063;
	setunitdata getnpcid(0),UNPC_HAIRSTYLE,1;
	setunitdata getnpcid(0),UNPC_BODY2,1;
	end;
}

`Program received signal SIGSEGV, Segmentation fault.
0x000000000046aa8f in clif_changelook (bl=0x7fffe2775a64, type=type@entry=13,
val=1) at clif.cpp:3704
warning: Source file is more recent than executable.
3704 if (val && sd->sc.option&OPTION_COSTUME)
Missing separate debuginfos, use: debuginfo-install glibc-2.17-260.el7_6.6.x86_64 libgcc-4.8.5-36.el7_6.2.x86_64 libstdc++-4.8.5-36.el7_6.2.x86_64 pcre-8.32-17.el7.x86_64 zlib-1.2.7-18.el7.x86_64
(gdb) bt full
#0 0x000000000046aa8f in clif_changelook (bl=0x7fffe2775a64,
type=type@entry=13, val=1) at clif.cpp:3704
sc = 0x7fffe2775f64
vd = 0x7fffe2775f44
sd = 0x0
target = AREA
val2 = 0
#1 0x0000000000588cb2 in buildin_setunitdata (st=0x7fffef816614)
at script.cpp:18308
hd = 0x7fffe08e52fc
ed = 0x194
nd = 0x7fffe2775a64
value = <optimized out>
__FUNCTION__ = "buildin_setunitdata"

mc = 0x3
type = 43
mapname = <optimized out>
bl = 0x7fffe2775a64
md = 0x4bc0
pd = 0x562ea1 <pop_stack(script_state*, int, int)+529>
calc_status = false
#2 0x0000000000563b1d in run_func (st=st@entry=0x7fffef816614)
at script.cpp:4056
data = <optimized out>
i = <optimized out>
start_sp = <optimized out>
func = 404
end_sp = <optimized out>
#3 0x0000000000568b20 in run_script_main (st=0x7fffef816614)
at script.cpp:4331
c = C_FUNC
cmdcount = 655332
gotocount = 2048
sd = <optimized out>
stack = 0x7fffeb2eb44c
#4 0x00000000004eef03 in npc_click(map_session_data*, npc_data*) ()
No symbol table info available.
#5 0x000000000046724d in clif_parse (fd=16) at clif.cpp:21919
cmd = <optimized out>
packet_len = 7
sd = 0x63b6970
pnum = 0
#6 0x000000000062c6dd in do_sockets (next=<optimized out>) at socket.cpp:1398
rfd = {fds_bits = {65536, 0 <repeats 15 times>}}
timeout = {tv_sec = 0, tv_usec = 26304}
ret = 0
i = 16
#7 0x0000000000407bdd in main (argc=1, argv=0x7fffffffe5f8) at core.cpp:369
next = <optimized out>`

**Comment by ecdarreola** [comment]
In-conflict with #4448

**Comment by attackjom** [comment]
I have found the problem when use mob_avail to change sprite to JOB

It show the headless sprite.

Is this a  bug or clientside problem ?

**Comment by cahya1992** [comment]
> I have found the problem when use mob_avail to change sprite to JOB
> 
> It show the headless sprite.
> 
> Is this a bug or clientside problem ?

i dont have that issue with client 20180621ragexe
but i have an issue with Sprite Upper/Middle/Lower didn't show poperly for sprite weapon work
Screenshot_1

File: src/map/mob.cpp [hunk 1]
```diff
@@ -4348,55 +4348,296 @@ static int mob_read_sqldb(void)
...
+		std::string sex;
+
+		if (!this->asString(node, "Sex", sex))
+			return 0;
+
+		std::string sex_constant = "SEX_" + sex;
+
+		int constant;
+
+		if (script_get_constant(sex_constant.c_str(), &constant)) {
```
Comment (teededung) [inline src/map/mob.cpp]: Missing "!" before `(script_get_constant(sex_constant.c_str(), &constant)`

**Pull request merged**